app/controllers/shipit/api/commits_controller.rb

```diff
@@ -6,6 +6,12 @@ class CommitsController < BaseController
       def index
         render_resources stack.commits.reachable.includes(:statuses)
       end
+
+      def undeployed
+        stack.undeployed_commits do |undeployed_commits|
+          render_resources undeployed_commits
```

Comment:
That block API isn't really meant to be used this way. You will get commits in reverse order etc.

You might be better to just get the query you want: https://github.com/Shopify/shipit-engine/blob/20f7c28874fc9b5e29cb3ca6efee2471048d5f0b/app/models/shipit/stack.rb#L255

```ruby
stack.commits.reachable.newer_than(stack.last_deployed_commit).includes(:statuses)
```

So basically the same thing already in `index` but with one more criteria.

Reply:
If we are already doing this in index then we probably should change it to add a param to index for deployed vs not deployed as you mentioned above.